pkg/monitor/azure/nsg/nsg.go

```diff
@@ -0,0 +1,168 @@
+package nsg
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/netip"
+	"strings"
+	"sync"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v2"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/metrics"
+	"github.com/Azure/ARO-RP/pkg/monitor/dimension"
+	"github.com/Azure/ARO-RP/pkg/monitor/emitter"
+)
+
+const (
+	MetricInvalidDenyRule          = "monitor.preconfigurednsg.invaliddenyrule"
+	MetricSubnetAccessResponseCode = "monitor.preconfigurednsg.subnetaccessresponsecode"
+	MetricSubnetAccessForbidden    = "monitor.preconfigurednsg.subnetaccessforbidden"
+	MetricUnsuccessfulFPCreation   = "monitor.preconfigurednsg.fpcreationunsuccessful"
+	MetricNSGMonitoringTimedOut    = "monitor.preconfigurednsg.monitoringtimedout"
+)
+
+var expandNSG = "NetworkSecurityGroup"
+
+// NSGMonitor is responsible for performing NSG rule validations when preconfiguredNSG is enabled
+type NSGMonitor struct {
+	log     *logrus.Entry
+	emitter metrics.Emitter
+	oc      *api.OpenShiftCluster
+
+	wg *sync.WaitGroup
+
+	subnetClient *armnetwork.SubnetsClient
+	dims         map[string]string
+}
+
+func NewNSGMonitor(log *logrus.Entry, oc *api.OpenShiftCluster, subscriptionID string, subnetClient *armnetwork.SubnetsClient, emitter metrics.Emitter, wg *sync.WaitGroup) *NSGMonitor {
```

Comment:
We should force these to fail to compile if they do not implement the monitor interface. 

Please do the same for the cluster monitor.  

e.g.
```
var _ http.RoundTripper = (*tracerRoundTripper)(nil)
```

Reply:
done